### PR TITLE
Support GGML with CTransformers

### PR DESCRIPTION
**Pull request by karthikcs**
@PromtEngineer Please review PR for GGML Support with CTransformers

### CODE REVIEW

**Comment by PromtEngineer** [comment]
@karthikcs I tested the PR and its just generating random text. Can you test the model in a simple LLMChain and see if you get similar behavior.

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,7 @@
+SOURCE_DOCUMENTS_/
```
Comment (malikarun) [inline .gitignore:1]: I am wondering why `.gitignore` is removed in the first place. 

Aside, I think you are adding extra files in your local environment and trying to ignore them here. Which should not be the case.

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+*.bin
+*.pyc
+DB/
+test_chroma_ingest.py
+test_chroma.py
+SOURCE_DOCUMENTS/dress code.pdf
```
Comment (malikarun) [inline .gitignore:7]: Last 3 lines are another example of ignoring unnecessary files.

File: constants.py [hunk 3]
```diff
@@ -46,8 +46,11 @@
 # Select the Model ID and model_basename
 # load the LLM for generating Natural Language responses
 
-MODEL_ID = "TheBloke/Llama-2-7B-Chat-GGML"
-MODEL_BASENAME = "llama-2-7b-chat.ggmlv3.q4_0.bin"
+# MODEL_ID = "TheBloke/Llama-2-7B-Chat-GGML"
+# MODEL_BASENAME = "llama-2-7b-chat.ggmlv3.q8_0.bin"
+
+MODEL_ID = "TheBloke/vicuna-7B-v1.5-GGML"
+MODEL_BASENAME = "vicuna-7b-v1.5.ggmlv3.q8_0.bin"
```
Comment (malikarun) [inline constants.py:53]: Here you added a `GGML` model after removing the same type of model. Not required.

File: run_localGPT.py [hunk 4]
```diff
@@ -47,21 +47,11 @@ def load_model(device_type, model_id, model_basename=None):
...
-                "max_tokens": max_ctx_size,
-            }
-            if device_type.lower() == "mps":
-                kwargs["n_gpu_layers"] = 1000
-            if device_type.lower() == "cuda":
-                kwargs["n_gpu_layers"] = 1000
-                kwargs["n_batch"] = max_ctx_size
-            return LlamaCpp(**kwargs)
-
+            # Llamacpp no more supports GGML format - https://huggingface.co/TheBloke/Llama-2-13B-chat-GGML/discussions/14#:~:text=cpp%20is%20no%20longer%20compatible,As%20far%20as%20llama.
```
Comment (malikarun) [inline run_localGPT.py:50]: I think the existing code can be improved but the new code is removing some existing functionality. Moreover, if you want to keep using GGML you can use an older version of `llama-cpp` locally. It's better for others to go ahead and start using new formats.

**Review by malikarun — CHANGES_REQUESTED** [review]
Please reconsider the need of this change. Thanks

File: run_localGPT.py (hunk 4 above)
Comment (karthikcs) [reply run_localGPT.py:50]: OK thanks for the review..

**Pull request closed without merging**